python/tests/test_mixed_log_reader.py

```diff
@@ -220,10 +217,8 @@ def test_read_no_generate_index(self, data_path):
         expected_messages = [m for m in messages if isinstance(m, PoseMessage)]
 
         reader = MixedLogReader(str(data_path), generate_index=False)
```

Comment:
Recommend renaming this arg to `save_index`, but also this unit test doesn't really make sense anymore as written. If we want to change it to a test for the `save_index` arg, that's fine, but then we need to check that the index file does not exist after this call, and then that the check results call below still works correctly.